src/hooks/mocha.ts

```diff
@@ -0,0 +1,66 @@
+import { simple as walk } from "acorn-walk";
+import type { ESTree } from "meriyah";
+import { expressionFor } from ".";
+import { call_, identifier } from "../generate";
+import { EventEmitter } from "node:stream";
+import Recording from "../Recording";
+import { recording, start } from "../recorder";
+
+export function shouldInstrument(url: URL): boolean {
+  return url.pathname.endsWith("/mocha/lib/runner.js");
+}
+
+export function transform(program: ESTree.Program): ESTree.Program {
+  return patchRuntime(program);
+}
+
+const EVENT_TEST_BEGIN = 'test';
+const EVENT_TEST_PASS = 'pass';
+const EVENT_TEST_FAIL = 'fail';
+const EVENT_TEST_END = 'test end';
+
+interface Test {
+  title: string
+  fullTitle: () => string
+}
```

Comment:
I think it's better to import  `@types/mocha` as a dev dependency than defining it here piecewise. It wouldn't bloat our runtime dependencies but will make sure the types are robust.

Reply:
Done. 

In `helpers.ts:integrationTest`, test identifier is resolved as a Mocha global variable instead of jest, at least in Visual Studio Code editor.

I tried to fix it with installing `@jest/globals` as dev dependency and imoırting test identifier explicitly but then, some other types were incompatible in static checking, like the type of `fn` argument. I removed `@jest/globals`. I'll go on based on your feedback.

<img width="747" alt="image" src="https://github.com/getappmap/appmap-node/assets/98758224/2ad91dc1-aa5b-462e-980e-68f2276d8462">

Reply:
Thanks, I'll take a look and see if I can fix that.

Reply:
Ugh, I can't get it to work, tsc seems to insist on importing these types even when not explicitly requested to do so. Never mind, I'll re-add the local definitions. Sorry, I really thought it would work :/